File: src/dataset/go/minesweeper_Exercism_go_rootulp/claude/ClaudeSonnet4_minesweeper_v4.go
package minesweeper

import (
	"strconv"
)

// Annotate returns an annotated board
func Annotate(board []string) []string {
	if len(board) == 0 || len(board[0]) == 0 {
		return board
	}

	rows := len(board)
	cols := len(board[0])
	
	result := make([]string, rows)
	for i := 0; i < rows; i++ {
		result[i] = board[i]
	}
	
	resultBytes := make([][]byte, rows)
	for i := 0; i < rows; i++ {
		resultBytes[i] = []byte(result[i])
	}
	
	for row := 0; row < rows; row++ {
		for col := 0; col < cols; col++ {
			if board[row][col] == '*' {
				continue
			}
			
			mines := 0
			for r := max(0, row-1); r <= min(rows-1, row+1); r++ {
				for c := max(0, col-1); c <= min(cols-1, col+1); c++ {
					if r == row && c == col {
						continue
					}
					if board[r][c] == '*' {
						mines++
					}
				}
			}
			
			if mines > 0 {
				resultBytes[row][col] = byte('0' + mines)
			}
		}
	}
	
	for i := 0; i < rows; i++ {
		result[i] = string(resultBytes[i])
	}
	
	return result
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

File: src/dataset/typescript/minesweeper_Exercism-typescript-shybyte/claude/ClaudeSonnet4_minesweeper_v4.ts
function count3x3Window(board: string[], centerX: number, centerY: number): number {
    let sum = 0;
    const boardHeight = board.length;
    const rowWidth = board[0]?.length || 0;
    
    const startY = Math.max(0, centerY - 1);
    const endY = Math.min(boardHeight - 1, centerY + 1);
    const startX = Math.max(0, centerX - 1);
    const endX = Math.min(rowWidth - 1, centerX + 1);
    
    for (let y = startY; y <= endY; y++) {
        const row = board[y];
        for (let x = startX; x <= endX; x++) {
            if (row[x] === '*') {
                sum++;
            }
        }
    }
    return sum;
}

export default class Minesweeper {
    annotate(board: string[]): string[] {
        if (board.length === 0) return board;
        
        const result: string[] = new Array(board.length);
        
        for (let y = 0; y < board.length; y++) {
            const row = board[y];
            let newRow = '';
            
            for (let x = 0; x < row.length; x++) {
                if (row[x] === ' ') {
                    const count = count3x3Window(board, x, y);
                    newRow += count || ' ';
                } else {
                    newRow += row[x];
                }
            }
            result[y] = newRow;
        }
        
        return result;
    }
}